### ISSUE TITLE
Settings sync - Cannot dismiss "Configure..." quick pick dialog

### ISSUE THREAD

**Issue by lszomoru** [microsoft/vscode#99081]
Testing #98962

* Click `Configure...` action in the `Preferences sync` toolbar
* Click anywhere else in the workbench

Expected: Quick pick dialog should hide (same as the command palette)
Actual: Only way to dismiss the quick pick dialog is using the `Ok` button

**Comment by sandy081** [comment]
You can also use `Esc` to dismiss. 

Since this is for configuring a feature this needs to be stable and do not get dismissed easily. This is not new and we use them for such features.

**Comment by lszomoru** [comment]
I see. So why is hitting `Esc` an acceptable way to dismiss this dialog but clicking somewhere is not?

**Comment by sandy081** [comment]
I think it is more common that we click somewhere unknowingly and we still do not want the UI pieces to be lost. But `esc` has stronger semantics of dismissing.

**Comment by Tyriar** [comment]
@misolori what do you think of a `x` icon when hovering quick pick headers that only dismiss right now with escape? It's best if features are accessible to both keyboard and mouse (without needing an onscreen keyboard).

![image](https://user-images.githubusercontent.com/2193314/83545573-42ccad80-a4b4-11ea-8e02-4563465ff9ec.png)

**Comment by misolori** [comment]
Clicking outside of the quick pick should also close this dialog, that's how the quick pick pattern works. Are we preventing this from happening? I don't mind the close button but it's harder to do and be consistent in other quick picks where there is no title.

**Comment by misolori** [comment]
We've previously ran into issues where the default behavior changes for various reasons but the user has no idea why it changed. See https://github.com/microsoft/vscode/issues/82797 where we did the same thing for the branch picker and users found it confusing.. We also have a setting `workbench.quickOpen.closeOnFocusLost` that allows users to define this preference. IMO, we shouldn't change this behavior and keep it consistent.

**Comment by Tyriar** [comment]
> Clicking outside of the quick pick should also close this dialog, that's how the quick pick pattern works. Are we preventing this from happening? I don't mind the close button but it's harder to do and be consistent in other quick picks where there is no title.

This is one of the special ones that prevents clicking away (or alt tabbing) from dismissing the dialog, so the only way to dismiss is escape or pressing OK.

**Comment by sandy081** [comment]
Yes, we allow non-dismissible quick picks for special use cases like this and others like

- file dialog
- auth account picker
- tasks quick pick
- Sync UI (turn on, configure)

Since this is going in a direction of having dismiss button for non dismissible quick picks assigning to @chrmarti

**Comment by misolori** [comment]
In the case where we really need non-dismissible quick picks, then showing a close button makes sense since those quick picks already have titles. I wonder if we can should enforce the close button to show up whenever the quick pick is prevented from closing?

**Comment by chrmarti** [comment]
Only a few QuickPicks prevent closing by clicking elsewhere. It is impossible for the user to anticipate / understand that difference. To improve that we could just not use that flag anymore ourselves (it's also part of the extension API, so we can't just remove its implementation).

On hovering: QuickPick supports a toolbar where the suggested close button would be. The toolbar items are always shown, I imagine the close button would have to do the same.

**Comment by sandy081** [comment]
> I'm not sure we need it with the configure action when sync is already on.

I am fine doing that.

**Comment by gjsjohnmurray** [comment]
@TylerLeonhardt this issue came up with the new Layout quickpick here:

https://github.com/microsoft/vscode/issues/137633#issuecomment-1020022014

Looks like it got solved by explicit addition of a close button. But it'd be nice if your proposal got implemented so that's no longer necessary to do.

**Comment by misolori** [comment]
I think if you can't close the quick pick via mouse or keyboard (independently) then that is an accessibility issue. So I'd suggest showing a close button when `ignoreFocusOut` is true, regardless of the titlebar.

**Comment by TylerLeonhardt** [comment]
@misolori you'd want a close button next to the input? We could also force the title bar to show up with the `x` if `ignoreFocusOut` is set.

**Comment by misolori** [comment]
That's definitely an option ,would look something like this for the branch picker that has this enabled:

<img width="690" alt="CleanShot 2022-02-01 at 20 23 05@2x" src="https://user-images.githubusercontent.com/35271042/152092830-472e4d0f-cf28-4bf8-acea-f75c1758c430.png">


An alternative would be to place a close button inside of the text input. We already use actions inside of inputs (find toggles, clearing search in settings, etc.) so this wouldn't stretch too far from our patterns, though we do use the "clear-all" icon and someone could confuse the "close" with that

<img width="761" alt="CleanShot 2022-02-01 at 20 20 44@2x" src="https://user-images.githubusercontent.com/35271042/152092683-a79d95a7-ee3f-44a8-8205-b58928eef152.png">

**Comment by gjsjohnmurray** [comment]
The new history item reference picker of the SCM Graph view also suffers from the lack of a way to dismiss it using the mouse. @lszomoru maybe you could lobby to get this issue addressed generically rather than simply adding your own.
